src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/ConfigurationTests.cs

```diff
@@ -147,6 +167,149 @@ public void ConfigurationOrderingTest(ConfigurationLevel level)
+            using TemporaryDirectory sharedConfigDir = new(_outputHelper);
+            using TemporaryDirectory userConfigDir = new(_outputHelper);
+
+            // Set up the initial settings used to create the host builder.
+            HostBuilderSettings settings = new()
+            {
+                Authentication = HostBuilderHelper.CreateAuthConfiguration(noAuth: false, tempApiKey: false),
+                ContentRootDirectory = contentRootDirectory.FullName,
+                SharedConfigDirectory = sharedConfigDir.FullName,
+                UserConfigDirectory = userConfigDir.FullName
+            };
+
+            // This is the settings.json file in the user profile directory.
+            string userSettingsContent = File.ReadAllText(userSettingsFilePath);
+            File.WriteAllText(Path.Combine(userConfigDir.FullName, "settings.json"), userSettingsContent);
+
+            // Create the initial host builder.
+            IHostBuilder builder = HostBuilderHelper.CreateHostBuilder(settings);
+
+            // Override the environment configurations to use predefined values so that the test host
+            // doesn't inadvertently provide unexpected values. Passing null replaces with an empty
+            // in-memory collection source.
+            builder.ReplaceAspnetEnvironment();
+            builder.ReplaceDotnetEnvironment();
+            builder.ReplaceMonitorEnvironment();
+
+            // Build the host and get the Urls property from configuration.
+            IHost host = builder.Build();
+            IConfiguration rootConfiguration = host.Services.GetRequiredService<IConfiguration>();
+
+            Stream stream = new MemoryStream();
+
+            using ConfigurationJsonWriter jsonWriter = new ConfigurationJsonWriter(stream);
+            jsonWriter.Write(rootConfiguration, full: !redact, skipNotPresent: false);
+            jsonWriter.Dispose();
+
+            stream.Position = 0;
+
+            using (var streamReader = new StreamReader(stream))
+            {
+                string configString = streamReader.ReadToEnd();
+
+                _outputHelper.WriteLine(configString);
+
+                CompareOutput(configString, expectedConfiguration);
+            }
+        }
+
+        private string ConstructUserSettingsJson()
+        {
+            string[] fileNames = Directory.GetFiles(Path.Combine(Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location), "SampleConfigurations"));
+
+            IDictionary<string, JsonElement> combinedFiles = new Dictionary<string, JsonElement>();
+
+            foreach (var fileName in fileNames)
+            {
+                IDictionary<string, JsonElement> deserializedFile = JsonSerializer.Deserialize<IDictionary<string, JsonElement>>(File.ReadAllText(fileName));
+
+                combinedFiles = combinedFiles.Union(deserializedFile).ToDictionary(pair => pair.Key, pair => pair.Value);
```

Comment:
While this is one line and convenient, I'd imagine it's very inefficient (I know, were only processing ~10 files, so overall it probably isn't too slow). This looks to be O(n^2) processing time because the earlier files are readded to new dictionaries ~n times where n is the number of files.

Try taking the `deserializedFile`, iterate it, and insert its key-value pairs into `combinedFiles` using `Dictionary<TKey, TValue>.Add(TKey, TValue)`. You can use the deconstructor syntax to more easily iterate `deserializedFile`, e.g.
```cs
foreach ((string key, JsonElement element) in deserializedFile)
{
    combinedFiles.Add(key, element);
}
```

Reply:
I had the same thought but didn't bother because of the small scale - I'll get this fixed in the next commit.